x-pack/legacy/plugins/apm/server/lib/helpers/convert_ui_filters/get_ui_filters_es.ts

```diff
@@ -9,12 +9,33 @@ import { ESFilter } from 'elasticsearch';
 import { UIFilters } from '../../../../typings/ui-filters';
 import { getEnvironmentUiFilterES } from './get_environment_ui_filter_es';
 import { getKueryUiFilterES } from './get_kuery_ui_filter_es';
+import {
+  localUIFilters,
+  localUIFilterNames
+} from '../../ui_filters/local_ui_filters/config';
 
 export async function getUiFiltersES(server: Server, uiFilters: UIFilters) {
-  const kuery = await getKueryUiFilterES(server, uiFilters.kuery);
-  const environment = getEnvironmentUiFilterES(uiFilters.environment);
+  const { kuery, environment, ...localFilters } = uiFilters;
+
+  const mappedFilters = localUIFilterNames
```

Comment:
I think I'll keep this one as-is. Indeed we need additional typing here (`value` is typed as `any`), but this is also less robust, because it's really hard to guarantee (even with a type system) that `name` is a valid filter name. Given that we would have to explicitly throw here + add type casts I think the current solution is better.